src/sources/fluent/mod.rs

```diff
@@ -92,8 +102,50 @@ impl SourceConfig for FluentConfig {
         )
     }
 
-    fn outputs(&self, _global_log_namespace: LogNamespace) -> Vec<Output> {
-        vec![Output::default(DataType::Log)]
+    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
+        // `host_key` is only inserted if not present already.
+        let host_key = parse_value_path(log_schema().host_key())
+            .ok()
+            .map(LegacyKey::InsertIfEmpty);
+
+        let tag_key = parse_value_path("tag").ok().map(LegacyKey::Overwrite);
+
+        let log_namespace = global_log_namespace.merge(self.log_namespace);
+
+        // There is a global and per-source `log_namespace` config.
+        // The source config overrides the global setting and is merged here.
+        let mut schema_definition = BytesDeserializerConfig
+            .schema_definition(log_namespace)
+            .with_standard_vector_source_metadata()
+            .with_source_metadata(
+                FluentConfig::NAME,
+                host_key,
+                &owned_value_path!("host"),
+                Kind::bytes(),
+                Some("host"),
+            )
+            .with_source_metadata(
+                FluentConfig::NAME,
+                tag_key,
+                &owned_value_path!("tag"),
+                Kind::bytes(),
+                Some("tag"),
```

Comment:
```suggestion
                None,
```

I've only been setting a meaning, when it's `host` as we really haven't explored this anywhere except for the `datadog_agent`.

Reply:
ok, I'll adapt to that convention. Not exactly clear on what "meaning" ... means :laughing:

Reply:
Yeah, I'd say it's not clearly defined at any point - so not something you're really "missing"